PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx

```diff
@@ -242,6 +241,7 @@ struct HfCorrelatorDplusHadrons {
                    aod::TracksWDca const& tracks,
                    soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi> const& candidates, aod::BCsWithTimestamps const&)
   {
+    BinningType corrBinning{{binsZVtx, binsMultiplicity}, true};
```

Comment:
Can you make this a member of the struct and initialise it only once in `init`?

Reply:
Hi @vkucera I was thinking this option before and tried also. When I initilaize there then it fails compilation also tried now. Do you have any example for this ? 
[compile.log](https://github.com/AliceO2Group/O2Physics/files/14097320/compile.log)

Reply:
```
BinningType corrBinning; // declared in stuct
corrBinning{{binsZVtx, binsMultiplicity}, true}; // initialized in the init function
```
Again getting the compilation error. May be definition in Init needs to be changed?

Reply:
Thanks, I need to understand looking at the example, I tried a few ways but not working at the moment.

Reply:
Thanks for the update.
Did you test that it works with different binning provided from JSON?
Can you do the same with `corrBinningMcGen` too?

Reply:
No in the previous case I tested it was working. After changing it I just checked for compilation (I will check for json also).  For MCGen, it's different we need getTracksSize which is a local function.
```
    using BinningTypeMCGen = FlexibleBinningPolicy<std::tuple<decltype(getTracksSize)>, aod::mccollision::PosZ, decltype(getTracksSize)>;
    BinningTypeMCGen corrBinningMcGen{{getTracksSize}, {binsZVtx, binsMultiplicityMc}, true};
    ```

Reply:
I checked the definition from json file and it works fine. For the Mc, I already explained the reason.

Reply:
Thanks